doc/source/whatsnew/v1.6.0.rst

```diff
@@ -213,6 +213,7 @@ Styler
 
 Other
 ^^^^^
+- Bug in :func:`pandas.infer_freq`, raising ``TypeError`` when inferred on :class:`RangeIndex` (:issue:`47084`)
```

Comment:
Could you put this in the `Datetimelike` section?

Reply:
Done! :)